test_conformance/subgroups/subhelpers.h

```diff
@@ -354,25 +1233,55 @@ struct test
         clProgramWrapper program;
         clKernelWrapper kernel;
         cl_platform_id platform;
-        cl_int sgmap[2 * GSIZE];
+        cl_int sgmap[4 * GSIZE];
```

Comment:
I believe some kernels where not updated to use `cl_int4` instead of `cl_int2`. For example, in `test_conformance/subgroups/test_barrier.cpp`. These kernels should also use the `XY` macro for consistency and decrease maintenance complexity.

(Github doesn't allow me to comment on lines that weren't changed by the PR...)